src/Controller/DefaultController.php

```diff
@@ -41,16 +46,15 @@ public function admin()
     /**
      * Get the sitemap.xml containing all dataset urls.
      *
-     * @return Response
+     * @Route("/sitemap.xml", name="pelagos_sitemap")
+     *
+     * @return StreamedResponse
      */
     public function showSiteMapXml()
     {
-        $container = $this->container;
-        $response = new StreamedResponse(function () use ($container) {
-
-            $entityManager = $container->get('doctrine.orm.entity_manager');
+        $response = new StreamedResponse(function () {
 
-            $datasets = $entityManager->getRepository(Dataset::class)->findBy(
+            $datasets = $this->getDoctrine()->getRepository(Dataset::class)->findBy(
```

Comment:
No injection for the EM here?

Reply:
It is just another way to do it, but probably we should come up with a set of guidelines when we should inject and when we can use these symfony injected services